### PR TITLE
Fix golint failures of kubeadm/v1beta1/...

### PR DESCRIPTION
**Pull request by xichengliudui**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Fix golint failures of kubeadm/v1beta1/...

**Which issue(s) this PR fixes**:

Ref #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/apis/kubeadm/v1beta1/defaults.go [hunk 1]
```diff
@@ -131,24 +131,25 @@ func SetDefaults_Etcd(obj *ClusterConfiguration) {
...
 	}
 
-	SetDefaults_JoinControlPlane(obj.ControlPlane)
-	SetDefaults_Discovery(&obj.Discovery)
+	SetDefaultsJoinControlPlane(obj.ControlPlane)
+	SetDefaultsDiscovery(&obj.Discovery)
 }
 
-func SetDefaults_JoinControlPlane(obj *JoinControlPlane) {
+// SetDefaultsJoinControlPlane when controlplane is not null, the native API port is used
```
Comment (yagonobre) [inline cmd/kubeadm/app/apis/kubeadm/v1beta1/defaults.go]: Maybe you can use something like `SetDefaultsJoinControlPlane assigns default values for the API server` wdyt?

**Review by yagonobre — COMMENTED** [review]
Thanks @xichengliudui 
/priority backlog

File: cmd/kubeadm/app/apis/kubeadm/v1beta1/defaults.go (hunk 1 above)
Comment (xichengliudui) [reply cmd/kubeadm/app/apis/kubeadm/v1beta1/defaults.go]: ok, thanks.

**Comment by xichengliudui** [comment]
Are these thrown errors because I still need to modify the `https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta1/zz_generated.defaults.go`.
Is it because it calls these functions?

**Review by neolit123 — COMMENTED** [review]
@xichengliudui 
thanks for the PR

please run:
./hack/verify-golint.sh
./hack/verify-govet.sh
./hack/verify-generated-files.sh

there is also `./hack/update-all.sh` to update generated files.

**Comment by neolit123** [comment]
actually i don't think you shouldn't touch `cmd/kubeadm/app/apis/kubeadm/v1beta1`...
to my knowledge we have to ignore it for golint because we cannot fix some of the lint problems in there.

/assign @rosti @fabriziopandini

**Comment by xichengliudui** [comment]
I'm sorry, I didn't know that. Is there anything you need me to do?

**Review by rosti — COMMENTED** [review]
@xichengliudui thanks for this!
I like the idea of fixing the linter for the code that is not generated. However, most of the touched APIs are public ones and users of the package may actually use them directly somehow (even though they are not supposed to).
We can certainly do that for a newer version of the config though. Fixing the golint problems for non-generated files is enough to remove us from `hack/.golint_failures` (verify-golint ignores generated files).
@fabriziopandini @timothysc WDYT?

File: cmd/kubeadm/app/apis/kubeadm/v1beta1/defaults_windows.go [hunk 2]
```diff
@@ -21,6 +21,6 @@ package v1beta1
 const (
 	// DefaultCACertPath defines default location of CA certificate on Windows
 	DefaultCACertPath = "C:/etc/kubernetes/pki/ca.crt"
-	// DefaultUrlScheme defines default socket url prefix
-	DefaultUrlScheme = "tcp"
+	// DefaultURLScheme defines default socket url prefix
+	DefaultURLScheme = "tcp"
```
[the remark this answers is not in the document]
Comment (neolit123) [reply cmd/kubeadm/app/apis/kubeadm/v1beta1/defaults_windows.go]: we can't do these renames these in v1beta1 at this point.
possibly in v1beta2 with an action-required..

[the remark this answers is not in the document]
Comment (xichengliudui) [reply cmd/kubeadm/app/apis/kubeadm/v1beta1/defaults_windows.go]: Sorry, I shouldn't have changed this.

**Pull request closed without merging**